### ISSUE TITLE
nullable: true dons't output  'NULL' in mysql

### ISSUE THREAD

**Issue by fanybook** [typeorm/typeorm#1388]
when i option 'nullable: true'

```
    @Column({ type: "timestamp", nullable: true })
    created_at: timestamp;

    @Column({ type: "timestamp", nullable: true })
    updated_at: timestamp;

    @Column({ type: "timestamp", nullable: true })
    deleted_at: timestamp;
```

the sql is ' CREATE TABLE `users` (`id` int(11) NOT NULL PRIMARY KEY AUTO_INCREMENT, `name` varchar(255) NOT NULL, `created_at` timestamp, `updated_at` timestamp, `deleted_at` timestamp) ENGINE=InnoDB'

the result of execute is:
created_at -> NOT NULL DEFAULT CURRENT_TIMESTAMP
updated_at -> NOT NULL DEFAULT '0000-00-00 00:00:00'
deleted_at -> NOT NULL DEFAULT '0000-00-00 00:00:00'

if 'nullable: true' can output  'NULL' in mysql, like:
'CREATE TABLE `users` (`id` int(11) NOT NULL PRIMARY KEY AUTO_INCREMENT, `name` varchar(255) NOT NULL, `created_at` timestamp NULL, `updated_at` timestamp NULL, `deleted_at` timestamp NULL) ENGINE=InnoDB'

the result of execute is okay.

**Comment by pleerock** [comment]
I don't understand what is an issue and what did you try to fix. If there is no `NOT NULL` in column definition, its `NULL` by default, why there is a need to explicitly set `NULL` keyword?

**Comment by fanybook** [comment]
my database is 5.7.x

if without NULL keyword, the result of execute is:

```
created_at NOT NULL DEFAULT CURRENT_TIMESTAMP
updated_at NOT NULL DEFAULT '0000-00-00 00:00:00'
deleted_at NOT NULL DEFAULT '0000-00-00 00:00:00'
```

**Comment by pleerock** [comment]
I think you got something wrong. 

With `nullable: true` its:

```sql
created_at NOT NULL DEFAULT CURRENT_TIMESTAMP
updated_at NOT NULL DEFAULT '0000-00-00 00:00:00'
deleted_at NOT NULL DEFAULT '0000-00-00 00:00:00'
```
With `nullable: false` its:

```sql
created_at DEFAULT CURRENT_TIMESTAMP
updated_at DEFAULT '0000-00-00 00:00:00'
deleted_at DEFAULT '0000-00-00 00:00:00'
```

which is correct, right? When `NOT NULL` is not set in the query, the default is `NULL`, you don't need to provide `NULL` keyword.

**Comment by pleerock** [comment]
@daniel-lang @bartlomiej-korpus guys can anyone help me, I don't understand this issue

**Comment by bartlomiej-korpus** [comment]
@pleerock 
I tried to verify it. When using MySQL 5.5, if you don't provide nullability [NULL | NOT NULL] option in DDL, it defaults to NOT NULL.
Looks like @fanybook is on 5.5 and what he describes is default behaviour there.
It changed between versions and now in  MySQL 5.7 that TypeORM runs tests with its fine (defaults to NULL).
It isn't bug in per se TypeORM, its inconsistency in MySQL, I am not sure if we should account for it.

**Comment by pleerock** [comment]
It defaults to NOT NULL in all versions, right? I still did not get what is the issue.

**Comment by bartlomiej-korpus** [comment]
@pleerock it doesnt, defaults to NOT NULL in 5.5 but to NULL in 5.7. TypeORM expects NULL as default.

**Comment by AlexMesser** [comment]
There is several inconsistent things between MySQL 5.5 and 5.7 such as NULL's behaviour, no support for date precision and milliseconds, no support for `DEFAULT CURRENT_TIMESTAMP` for a `DATETIME` type and possibly others that are not yet discovered. My opinion is to drop support for this version to avoid unnecessary complexity in the implementation.

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.41`. You can try it via `npm i typeorm@next`.

**Comment by rmainseas** [comment]
@fanybook your comment(https://github.com/typeorm/typeorm/issues/1388#issuecomment-354733834) about a nullable column defaulting to `NULL` helped me. 

I was going crazy trying to figure out why TypeORM was repeatedly trying to set that in my migrations. I removed `default: null` from my entity and it immediately stopped Thank you.
